rekolekcje-engine/src/verify/java/pl/oaza/warszawa/dor/rekolekcje/api/participants/utils/ParticipantFactory.java

```diff
@@ -103,5 +133,68 @@ public static ParticipantDTO participantWithAllData(Long id) {
       .stepsTaken(4)
       .stepsPlannedThisYear(6)
       .deuterocatechumenateYear(2016)
+      .historicalRetreats(historicalTurns)
       .build();
+
+  public static ParticipantDTO copyWithDifferentId(ParticipantDTO original, long participantId, Set<RetreatTurnValue> retreatsWithIds) {
+    final ExperienceValue newExperienceValue = copyExperienceWithRetreats(original.getExperience(), retreatsWithIds);
+    return ParticipantDTO.builder()
+        .id(participantId)
+        .firstName(original.getFirstName())
+        .lastName(original.getLastName())
+        .pesel(original.getPesel())
+        .parishId(original.getParishId())
+        .personalData(original.getPersonalData())
+        .address(original.getAddress())
+        .experience(newExperienceValue)
+        .healthReport(original.getHealthReport())
+        .build();
+  }
+
+  private static ExperienceValue copyExperienceWithRetreats(ExperienceValue original, Set<RetreatTurnValue> newRetreats) {
+    return ExperienceValue.builder()
+        .historicalRetreats(newRetreats)
+        .kwcStatus(original.getKwcStatus())
+        .kwcSince(original.getKwcSince())
+        .numberOfCommunionDays(original.getNumberOfCommunionDays())
+        .numberOfPrayerRetreats(original.getNumberOfPrayerRetreats())
+        .deuterocatechumenateYear(original.getDeuterocatechumenateYear())
+        .stepsTaken(original.getStepsTaken())
+        .stepsPlannedThisYear(original.getStepsPlannedThisYear())
+        .celebrationsTaken(original.getCelebrationsTaken())
+        .celebrationsPlannedThisYear(original.getCelebrationsPlannedThisYear())
+        .formationMeetingsInMonth(original.getFormationMeetingsInMonth())
+        .leadingGroupToFormationStage(original.getLeadingGroupToFormationStage())
+        .build();
+  }
+
+  public static ParticipantDTO withNewData(long id) {
```

Comment:
To samo co wyżej. Dużo kodu jest bardzo podobnego jak przy tworzeniu przykładowych DTO uczestnikow

Reply:
Jak będę miał czas to uporządkuję trochę tę klasę. Na razie chodzi głównie o to, co napisałem powyżej. Nie chcemy mieć setterów i chcemy operować na w miarę niemutowalnych obiektach, dlatego takie kopiowanie. Docelowo może stworzę jakąś taką bardziej "fabryczną" fabrykę (w sensie bliższą temu wzorcowi projektowemu), a raczej hierarchię fabryk, które będą produkowały różnego rodzaju obiekty, i wtedy całość się posprząta.